src/services/exit_order.py

```diff
@@ -1,24 +1,335 @@
+                )
+                if on_exit or previously_requested_to_exit:
+                    # We already counted this validator in initial values (total_deposited_validators).
+                    # Need to decrease it if it's on exit
+                    operator_predictable_states[module_operator].predictable_validators_count -= 1
+                    continue
+                validator_age = max(0, blockstamp.ref_epoch - int(validator.validator.validator.activation_epoch))
+                operator_predictable_states[module_operator].predictable_validators_total_age += validator_age
+
+        return operator_predictable_states
+
+    def _get_last_requested_to_exit_indices(
+        self, blockstamp: BlockStamp, operator_indexes: Iterable[NodeOperatorIndex],
+    ) -> dict[NodeOperatorIndex, int]:
+        """
+        Get last requested to exit validator index for each operator
+        """
+        module_operator_ids = defaultdict(set)
+        for module_id, operator_id in operator_indexes:
+            module_operator_ids[module_id].add(operator_id)
+
+        last_requested_to_exit_indexes = {}
+        for module_id, operator_ids in module_operator_ids.items():
+            per_operator_indexes = self._get_last_requested_validator_index(blockstamp, module_id, list(operator_ids))
+            for array_index, operator_id in enumerate(operator_ids):
+                last_requested_to_exit_indexes[(module_id, operator_id)] = per_operator_indexes[array_index]
+        return last_requested_to_exit_indexes
+
+    def _get_delayed_validators_count_per_operator(
+        self,
+        operator_validators: dict[NodeOperatorIndex, list[LidoValidator]],
+        recently_requested_to_exit_indices_per_operator: dict[NodeOperatorIndex, set[int]],
+        last_requested_to_exit_indices_per_operator: dict[NodeOperatorIndex, int],
+    ) -> dict[NodeOperatorIndex, int]:
+        """
+        Get delayed validators count for each operator
+        """
+
+        delayed_validators_count = defaultdict(int)
+
+        for module_operator, validators in operator_validators.items():
+            recently_requested_to_exit_indices = recently_requested_to_exit_indices_per_operator[module_operator]
+            for validator in validators:
+                previously_requested_to_exit = (
+                    int(validator.validator.index) <= last_requested_to_exit_indices_per_operator[module_operator]
+                )
+                on_exit = self._is_on_exit(validator.validator)
+                recently_requested_to_exit = int(validator.validator.index) in recently_requested_to_exit_indices
+                if previously_requested_to_exit and not on_exit and not recently_requested_to_exit:
+                    delayed_validators_count[module_operator] += 1
+
+        return delayed_validators_count
+
+    def _get_recently_requested_to_exit_indices(
```

Comment:
Use get_events_in_past method in utils folder

Reply:
It `_get_recently_requested_to_exit_indices` method we need raw events without any filtering and so on because it has its own filtering logic